discord/ext/ipcx/client.py

```diff
@@ -151,4 +152,12 @@ async def request(self, endpoint: str, **kwargs) -> Any:
 
                 return await self.request(endpoint, **kwargs)
 
+            if recv.type == aiohttp.WSMsgType.CLOSE:
+                close_code = recv.data
+                reason = WSCloseCode(close_code).name
+
+                raise WebSocketClosedError(
+                    f"WebSocket connection unexpectedly closed with code: {close_code} ({reason})"
+                )
+
```

Comment:
No need for this logic as you kill the connection without attempting to reconnect. This will lead to more issues in the future. Alternatively, just merge the check with the `if` statement above

Reply:
all of the error handling I added can be removed, as it has nothing to do specifically with the max size issue. It's just that currently, the errors are straight up misleading without digging. All I added was to raise an error on the server side if an error occurs (without this, an error will still be raise, it will just be `TypeError: the JSON object must be str, bytes or bytearray, not WebSocketError`, which isn't exactly helpful). And on the client side, if the error is a close error, then again, a more meaningful message will be raised (otherwise, `TypeError: the JSON object must be str, bytes or bytearray, not int` would be raised, because the close code is in place of data, which gets fed into json.load and throws an error). I don't do anything additional with the connection, such as killing it. Just make the error more meaningful. All of that can be removed tho, especially if u r sure that no more such errors can occur (I haven't exactly examined much of the code, but I don't see much error handling)